eZ/Bundle/EzPublishCoreBundle/ApiLoader/SignalSlot/SignalDispatcherFactory.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+/**
+ * This file is part of the eZ Publish Kernel package.
+ *
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace eZ\Bundle\EzPublishCoreBundle\ApiLoader\SignalSlot;
+
+use eZ\Publish\Core\Base\Container\ApiLoader\SignalSlot\SignalDispatcherFactory as BaseSignalDispatcherFactory;
+
+class SignalDispatcherFactory extends BaseSignalDispatcherFactory
+{
+    /**
+     * SignalDispatcherFactory constructor.
+     *
+     * @param string $signalDispatcherClass
+     * @param string $repositoryAlias
+     * @param array $repositoriesSettings
+     */
+    public function __construct(
+        $signalDispatcherClass,
+        $repositoryAlias,
+        array $repositoriesSettings
+    ) {
+        if ($repositoryAlias === null) {
+            $aliases = array_keys($repositoriesSettings);
+            $repositoryAlias = array_shift($aliases);
+        }
+        $searchEngineAlias = isset($repositoriesSettings[$repositoryAlias]['search']['engine']) ? $repositoriesSettings[$repositoryAlias]['search']['engine'] : [];
```

Comment:
> I would rather avoid doing the resolving in the constructor, it's not a big problem as the logic is light, but it would still be better to have a separate method to do it on demand (particularly if we need to change search engine later on for example).

@pspanja: should I move this whole logic (lines 27-31) to `getSearchEngineAlias` (including resolving repository settings) or just store here single repository settings and then in `getSearchEngineAlias` only return `$this->repositorySettings['search']['engine']`?